### DISCUSSION TITLE
Yarp service discovery

### DISCUSSION THREAD

**Discussion by mlankamp** [microsoft/aspire#6674] [General]
I trying to put Yarp in front of my backend APIs, using the `Microsoft.Extensions.ServiceDiscovery.Yarp` package. I have setup my BFF as following:

```c#

var builder = WebApplication.CreateBuilder(args);

builder.AddServiceDefaults();

builder.Services.AddHttpForwarderWithServiceDiscovery();

var app = builder.Build();

app.MapForwarder("/service1/{**catch-all}", "http://service1");

app.MapForwarder("/service2/{**catch-all}", "http://service2");

app.MapForwarder("/service3/{**catch-all}", "http://service3");

app.MapForwarder("/{**catch-all}", "http://ui");

app.Run();

```

Locally this works perfectly, but when I deploy to Azure (with `azd up`) the UI works, but all the requests to backend services yarp responds with 301. This is correct, as the backend services in Azure are created with ingress setting "Allow Insecure = false".

One option is to use `azd config set alpha.infraSynth on` and `azd infra synth` and update the yaml files to set `allowInsecure=true`

How can I tell my MapForwarder to use the https url if available?

**Comment by davidfowl** [comment]
https://learn.microsoft.com/en-us/dotnet/core/extensions/service-discovery?tabs=dotnet-cli#scheme-selection-when-resolving-https-endpoints

e.g.

```C#
app.MapForwarder("/service1/{**catch-all}", "https+http://service1");
```

**Comment by mlankamp** [comment]
@davidfowl I don't think https is needed for my backend services as they are internal. Can I configure them as http(-only) in the AppHost?

When I read https://github.com/Azure/azure-dev/issues/3021 it looks like internal apps should allow http
